docs/proposals/job_controller.md

```diff
@@ -0,0 +1,191 @@
+# Job Controller
+
+## Abstract
+A proposal for implementing a new controller - Job controller, which will be responsible
+for managing pods that require to run:
+* once on demand or at a specified point in time
+* repeatedly at a specified point in time.
+
+Several exiting issues were already created regarding that particular subject:
+* Distributed CRON jobs [#2156](https://github.com/GoogleCloudPlatform/kubernetes/issues/2156)
+* Job Controller [#1624](https://github.com/GoogleCloudPlatform/kubernetes/issues/1624)
+
+
+## Use Cases
+1. Ability to have a controller that will create a run-once pod.
+1. Ability to have a controller that will have an option to specify reaction on pod’s success/failure.
+1. Ability to specify number of restarts upon pod failure.
+1. Ability to specify maximum running time for a pod.
```

Comment:
I think the maximum running time should be that of the Job which is materialized by N pods.  I worry that we are to strongly associating a Job as meaning 1 Pod.  For example, if I set a maximum running time of a Job as 10 minutes, and I kick off the Job, but no Pods are ever scheduled in 10 minutes after I kick off the Job, I view the Job as failed.  As a result, the running time is from the point that the JobController is actively trying to spawn pods, not the time that the pod is actually running on the host.